### PR TITLE
WAITP-1418: rewrite fixes

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Fix Matrix mobile screen warning
- Always use expanded state data for dashboard item reports
- Fix some styling issues
- Add mapbox attribution

### CODE REVIEW

File: packages/tupaia-web/src/features/Dashboard/DashboardMenu.tsx [hunk 1]
```diff
@@ -99,19 +100,18 @@ export const DashboardMenu = ({
...
       )}
-
-      <ItemButton
+      <StyledMenu
         id="dashboards-menu"
         anchorEl={anchorEl}
         open={Boolean(anchorEl)}
         onClose={handleClose}
         variant="menu"
-        disablePortal
```
Comment (tcaiger) [inline packages/tupaia-web/src/features/Dashboard/DashboardMenu.tsx:109]: fix z-index clash by making this a portal

File: packages/tupaia-web/src/features/DashboardItem/DashboardItemContent.tsx [hunk 2]
```diff
@@ -33,9 +33,6 @@ const DisplayComponents = {
 const getHasNoData = (report: DashboardItemReport, type: DashboardItemConfig['type']) => {
   // If there is no report, if means it is loading or there is an error, which is handled elsewhere
   if (!report) return false;
-  if (type === 'matrix') {
```
Comment (tcaiger) [inline packages/tupaia-web/src/features/DashboardItem/DashboardItemContent.tsx:36]: Leave it up to the matrix component to display the no data message

File: packages/tupaia-web-server/src/routes/LegacyDashboardReportRoute.ts [hunk 3]
```diff
@@ -20,6 +20,10 @@ export class LegacyDashboardReportRoute extends Route<LegacyDashboardReportReque
     const { query, ctx } = this.req;
     const { reportCode } = this.req.params;
 
-    return ctx.services.webConfig.fetchReport(reportCode, { legacy: 'true', ...query });
+    return ctx.services.webConfig.fetchReport(reportCode, {
+      legacy: 'true',
+      isExpanded: 'true', // Always get the data for expanded reports. The only time there is a difference is when the report is on matrix reports, and we don't show them when not expanded
```
Comment (acdunham) [inline packages/tupaia-web-server/src/routes/LegacyDashboardReportRoute.ts:25]: Thanks for fixing that, I don't know why I didn't think of this before

File: packages/tupaia-web/src/features/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -62,7 +62,7 @@ const StickyBar = styled.div<{
 }>`
   position: sticky;
   top: 0;
-  z-index: 1;
+  z-index: 10;
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Dashboard/Dashboard.tsx:65]: Classic z-index wars

**Review by acdunham — APPROVED** [review]
LGTM

**Pull request merged**